scripts/python/packages/parthenon_tools/parthenon_tools/movie2d.py

```diff
@@ -142,45 +200,97 @@ def plot_dump(
+        )
+        logger.warning(
+            "Try decreasing threads if you encounter any undefined behaviour"
+        )
+    # Create output director if does't exists
+    args.output_directory.mkdir(0o755, True, True)
+    logger.info(f"Total files to process: {len(args.files)}")
+
+    components = [0, 0]
     if args.tc is not None:
         components = args.tc
     if args.vc is not None:
         components = [0, args.vc]
-    dump_id = 0
-    debug_plot = False
-    for f in files:
-        data = read(f)
-        print(data)
-        q = data.Get(field, False, not debug_plot)
-        name = str(dump_id).rjust(4, "0") + ".png"
-        if debug_plot:
-            plot_dump(
-                data.xg,
-                data.yg,
-                q,
-                name,
-                True,
-                data.gid,
-                data.xig,
-                data.yig,
-                data.xeg,
-                data.yeg,
-                components,
+
+    _x = ProcessPoolExecutor if args.worker_type == "process" else ThreadPoolExecutor
+    with _x(max_workers=args.workers) as pool:
+        for dump_id, file_name in enumerate(args.files):
+            data = phdf(file_name)
+            if args.field not in data.Variables:
+                logger.error(
+                    f'No such field "{args.field}" in {file_name}. This will lead to stop further processing'
+                )
+                logger.info(f"Available fields: {data.Variables}")
+                ERROR_FLAG = True
+                break
+
+            q = data.Get(args.field, False, not args.debug)
+            stem = f"{args.prefix}_{str(dump_id).rjust(4, '0')}".strip()
+            stem = re.sub(r"^_", "", stem)
+            name = stem + ".png"
+            output_file = args.output_directory / name
+
+            # NOTE: After doing 5 test on different precision, keeping 2 looks more promising
+            current_time = format(round(data.Time, 2), ".2f")
+            if args.debug:
+                pool.submit(
+                    plot_dump,
+                    data.xg,
+                    data.yg,
+                    q,
+                    current_time,
+                    output_file,
+                    True,
+                    data.gid,
+                    data.xig,
+                    data.yig,
+                    data.xeg,
+                    data.yeg,
+                    components,
+                )
+            else:
+                pool.submit(
+                    plot_dump, data.xng, data.yng, q, current_time, output_file, True
+                )
+
+    if not ERROR_FLAG:
+        logger.info("All files are sent to the processor")
+
+        if not args.no_render:
+            logger.info(f"Generating {args.movie_format} movie")
+            input_pattern = args.output_directory / "*.png"
+            ffmpeg_cmd = f"ffmpeg -hide_banner -loglevel error -y -framerate {args.frame_rate} -pattern_type glob -i '{input_pattern}' "
```

Comment:
Sometimes systems have `libav` and not `ffmpeg`. So it might be worth adding an option to switch ``ffmpeg` for `libav`.

Reply:
I have never heard of libav, but will explore an use it. Thanks for suggesting @Yurlungur

Reply:
:+1: